### PR TITLE
[QUIZ-59] [QUIZ-58] feature: api create draft quizzfly

### PR DESCRIPTION
**Pull request by duchuyyyy**
# Pull Request types

- 🐛 Bug Fix
- 👍 Improvement
- ✨ New Feature
- 🎉 Release
- ♻️ Refactoring
- 🚿 Refactoring (Remove functions)
- 💚 Refactoring (Test or CI)
- 👕 Fix (Lint)
- 🆙 Update (Dependency packages)
- 🚀 Improvement (Performance)
- 👮 Improvement (Security)
- 🔒 Restriction (Restrict new feature)
- 👾 Others

# Description

# Checklist

- [ ] 1. Build successfully on local
- [ ] 2. Format all code files included in the PR
- [ ] 3. No more unconfirmed specs in the PR
- [ ] 4. Tested the normal case and error cases following specs
- [ ] 5. Is there a refactor to the old code? If yes, have you tested the old function and noted the impact yet?
- [ ] 6. Loops have a set length and correct termination conditions
- [ ] 7. There aren't any redundant or duplicate codes in PR (clean code)
- [ ] 8. There aren't any hard-coded in the code in PR
- [ ] 9. Optimization logic handling and optimizations SQL (for example, don't get redundant data or add an index to optimize performance, query only necessary fields...)
- [ ] 10. Update issue status and add PR link to the issue

# Evidence
(Screenshot or Video)

# Discussion

### CODE REVIEW

**Comment by LeTatTuan** [comment]
- chưa có dto mẫu khi call api m: 
image

File: src/modules/quizzfly/quizzfly.service.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    return quizzflys.map(quizzfly => InfoQuizzflyResDto.toInfoQuizzflyResponse(quizzfly, user));
+  }
+
+  async findById(quizzflyId: Uuid) {
+    const quizzfly = await this.quizzflyRepository.findOne({
+      where: { id: quizzflyId },
+    });
+    if (!quizzfly) {
+      throw new NotFoundException('Quizzfly not found');
```
Comment (LeTatTuan) [inline src/modules/quizzfly/quizzfly.service.ts:69]: - nên đặt file errorcode chung

File: src/modules/user/user.service.ts [hunk 2]
```diff
@@ -44,6 +44,17 @@ export class UserService {
...
   }
 
+  async findByUserId(id: Uuid) {
+    const user = await this.userRepository.findOne({
+      where: { id },
+      relations: ['userInfo'],
+    });
+    if (!user) {
+      throw new NotFoundException(`User was not found`);
+    }
```
Comment (LeTatTuan) [inline src/modules/user/user.service.ts:54]: - cũng nên đặt error code chung, tại dùng i18n,  với mấy câu điều kiện: có lớp Optional trong chỗ core/utils

File: src/modules/quizzfly/quizzfly.controller.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  @Put(':quizzflyId/settings')
+  @ApiAuth({
+    type: InfoDetailQuizzflyResDto,
+    summary: 'Setting quizzfly',
+  })
+  @ApiParam({
+    name: 'quizzflyId',
+    description: 'The UUID of the Quizzfly',
+    type: 'string',
+  })
```
[the remark this answers is not in the document]
Comment (duchuyyyy) [reply src/modules/quizzfly/quizzfly.controller.ts:37]: t test thì thấy không có

File: src/modules/quizzfly/entity/quizzfly.entity.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+
+@Entity('quizzfly', { schema: 'public' })
+export class QuizzflyEntity extends AbstractEntity {
```
[the remark this answers is not in the document]
Comment (duchuyyyy) [reply src/modules/quizzfly/entity/quizzfly.entity.ts:14]: cái này mình validation ở dto được mà, nhỡ sau yêu cầu thay đổi thì sao

File: src/modules/quizzfly/dto/response/info-quizzfly.response.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    dto.coverImage = quizzflyEntity.coverImage;
+    dto.quizzflyStatus = quizzflyEntity.quizzflyStatus;
+    dto.createdAt = quizzflyEntity.createdAt;
+
+    dto.userId = userEntity.id;
+    dto.username = userEntity.userInfo.username;
+    dto.avatar = userEntity.userInfo.avatar;
+
+    return dto;
+  }
```
[the remark this answers is not in the document]
Comment (duchuyyyy) [reply src/modules/quizzfly/dto/response/info-quizzfly.response.ts:47]: Nam kêu đẩy ra bằng cấp nhau

**Pull request merged**